Docs/Debugging/Debugging with Eclipse and pyOCD.md

```diff
@@ -0,0 +1,68 @@
+    1. Install **make** and add it to path. Make is available as one of the tools here: [http://gnuwin32.sourceforge.net/](http://gnuwin32.sourceforge.net/).
+1. Install python 2.7.9 or above
+  1. Windows Only - Make sure to select "Add to path" when installing.
+1. Install pyOCD by running "pip install pyocd".
+
+##Importing to Eclipse and building
+1. In the mbed online compiler, export the project you want to debug as **GCC (Arm Embedded)**.
+1. Extract the project archive exported from mbed.
+1. Open Eclipse.
+1. Click the **File** menu item, mouse over **New** and select **Makefile Project with Existing Code**.
+  1. Paste the location of your project into **Existing Code Location**.
+  1. **Project Name** should automatically populate. Update name here if desired.
+  1. In **Toolchain for Indexer Settings**, select **Cross ARM GCC**.
+  1. Click **Finish** to create the project.
+1. To set up building in Eclipse, go to the **Project** menu item, and select **Properties**.
+  1. Select **C/C++ Build**.
+  1. Unselect **Use default build command**.
+  1. For **Build command**, enter **make DEBUG=1**. The option "DEBUG=1" configures the make file to build optimized code, which makes debugging easier.
+  1. Click **Apply**, and then click **OK**.
+  <span class="images">![](Images/30_-_build_config.png)</span>
+1. Click the build button in Eclipse, and verify the project builds successfully.
+
+##Setting up debug configuration
+The last step before actually debugging is to set up the debug configuration. This only needs to be done once for per project. Once created, the configuration can be checked into revision control, allowing anyone else working on the project to use it without needing to go through setup again.
+
+1. On the **Run** menu item, select **Debug Configurations...** to bring up configurations.
+1. Right click on **GDB pyOCD Debugging** and select **New**.
+1. Open **Main** tab, and verify **Project** is correct and **C/C++ Application** correctly points to the .elf file.
+<span class="images">![](Images/10_-_debug_main_3.png)</span>
+1. Open **Debugger** tab.
+  1. In the **pyOCD Setup** box:
+    1. Check **Start pyOCD locally** if it isn't checked already.
+    1. Set **Executable:** to **pyocd-gdbserver** or the full path to your installation.
+    1. Set **GDB port:** to a free port. The default value of 3333 should work.
+  1. In the **GDB Client Setup** box:
+    1. Set **Executable** to **arm-none-eabi-gdb.exe**.
+    1. In **Commands:**, add **set mem inaccessible-by-default off** if it is not present.
+  1. It should look something like this when configured: 
+  <span class="images">![](Images/11_-_debug_debugger_2.png)</span>
+  1. Defaults will work in **Startup** tab.
+  1. Defaults will work in **Source** tab.
+  1. Open **Common** tab.
+    1. Under **Save as**, select **Shared file**. The default name will work. This allows the debug configuration to be saved locally and checked into the current project.
```

Comment:
"Allows to" implies I still need to do something to save it locally. I suspect I don't.